test/distributed/_tensor/test_random_ops.py

```diff
@@ -0,0 +1,127 @@
+
+import torch
+
+from torch.distributed._tensor import DeviceMesh, DTensor
+from torch.distributed._tensor.placement_types import Replicate, Shard
+from torch.distributed._tensor.random import _set_offset, get_rng_state, manual_seed
+
+from torch.distributed.distributed_c10d import broadcast_object_list
+
+from torch.testing._internal.common_utils import run_tests
+from torch.testing._internal.distributed._tensor.common_dtensor import (
+    DTensorTestBase,
+    skip_unless_torch_gpu,
+    with_comms,
+)
+
+
+class DistTensorRandomOpTest(DTensorTestBase):
+    @with_comms
+    def test_device_mesh_init(self):
+        # device mesh init should sync seed and store it as an attribute
+        object_list = [torch.cuda.initial_seed()]
+        broadcast_object_list(object_list)
+        seed_from_rank_0 = int(object_list[0])
+
+        device_mesh = DeviceMesh(self.device_type, torch.arange(self.world_size))
+        self.assertEqual(seed_from_rank_0, device_mesh._seed)
+
+    @with_comms
+    @skip_unless_torch_gpu
+    def test_manual_seed(self):
+        device_mesh = DeviceMesh(self.device_type, torch.arange(self.world_size))
+        manual_seed(1234, device_mesh)
+        with self.assertRaisesRegex(RuntimeError, "different seed values"):
+            manual_seed(self.rank, device_mesh)
+
+    @with_comms
+    @skip_unless_torch_gpu
+    def test_deterministic_uniform_1d(self):
+        def check_rng_state(seed: int, offset: int, device_mesh: DeviceMesh) -> None:
+            state = get_rng_state(device_mesh)
+            seed_int64 = state[-16:-8].view(torch.int64)
+            offset_int64 = state[-8:].view(torch.int64)
+            self.assertEqual(seed_int64, torch.tensor([seed]))
+            self.assertEqual(offset_int64, torch.tensor([offset]))
+
+        device_mesh = DeviceMesh(self.device_type, torch.arange(self.world_size))
+        size = [4, 1]
+
+        # initialize rng state
+        manual_seed(1234, device_mesh)
+        check_rng_state(1234, 0, device_mesh)
+        _tensor = torch.empty(*size, device="cuda")
+        dtensor = DTensor.from_local(_tensor, device_mesh, [Shard(1)])
+
+        # preprocess rng offset
+        global_size = dtensor.numel()
+        local_size = dtensor.to_local().numel()
+        state = get_rng_state(device_mesh)
+        offset = state[-8:].view(torch.int64)[0].item()
+        _set_offset(offset + self.rank * local_size, device_mesh)
+        check_rng_state(1234, self.rank * local_size, device_mesh)
+
+        # random op call
+        dtensor.uniform_(0, 1)
+
+        # postprocess rng offset
+        _set_offset(offset + global_size, device_mesh)
+        check_rng_state(1234, global_size, device_mesh)
+
+        # allgather the local tensors
+        local_tensor = dtensor.to_local()
+        local_tensor_list = [
+            torch.empty_like(local_tensor) for i in range(self.world_size)
+        ]
+        device_mesh.all_gather(local_tensor_list, local_tensor)
```

Comment:
does it work? can you rebase to the lastest mater? I think `mesh.allgather` changes its API to return a single bigger tensor instead of tensor list. So I think this test should crash. 

Can you just call `dtensor.redistribute([Replicate()]).to_local()` instead of manually call allgather here? we should try to use redistribute conversion instead of individual collective imo.

Reply:
replaced `allgather` call with a `redistribute([Replicate()])`.